### PR TITLE
SWATCH-2897 Internal api to provide remittance response based on tally_id

### PR DESCRIPTION
**Pull request by kartikshahc**
Jira issue: SWATCH-2897

## Testing

IQE Test MR: 
https://gitlab.cee.redhat.com/insights-qe/iqe-rhsm-subscriptions-plugin/-/merge_requests/861

### Setup

1.` INSERT INTO public.billable_usage_remittance (account_number, org_id, product_id, metric_id, accumulation_period, sla, usage, billing_provider, billing_account_id, remitted_pending_value, remittance_pending_date, retry_after, tally_id, hardware_measurement_type, uuid, billed_on, error_code, status) VALUES (null, '17795559', 'ansible-aap-managed', 'Managed-nodes', '2024-09', 'Premium', 'Production', 'aws', 'aws123', 2, '2024-09-16 16:25:56.198009 +00:00', null, '140a676c-86b0-4883-9b33-e6ac44dbef92', 'PHYSICAL', 'bb5afb2c-e2ea-4e56-9c9c-415b7641fa22', null, null, 'gratis');`

2. `SERVER_PORT=8002 QUARKUS_MANAGEMENT_PORT=9002 DEV_MODE=true ./gradlew clean :swatch-billable-usage:quarkusDev`

### Steps

1. `http://localhost:8002/api/swatch-billable-usage/internal/remittance/accountRemittances/140a676c-86b0-4883-9b33-e6ac44dbef92`

### Verification

1. 
```
[
{
"orgId": "17795559",
"productId": "ansible-aap-managed",
"metricId": "Managed-nodes",
"billingProvider": "aws",
"billingAccountId": "aws123",
"remittedPendingValue": 2,
"accumulationPeriod": "2024-09",
"remittancePendingDate": "2024-09-16T16:25:56.198009Z",
"status": "GRATIS",
"errorCode": null,
"tallyId": "d51d37f1-0ed2-441e-b59a-a6adeb8c1178"
}
]
```

### CODE REVIEW

File: swatch-billable-usage/build.gradle [hunk 1]
```diff
@@ -31,6 +31,11 @@ dependencies {
     implementation project(":swatch-model-billable-usage")
     annotationProcessor enforcedPlatform(libraries["quarkus-bom"])
     annotationProcessor "org.hibernate.orm:hibernate-jpamodelgen"
+    implementation libraries["mapstruct"]
+    annotationProcessor libraries["mapstruct-processor"]
+    testAnnotationProcessor libraries["mapstruct-processor"]
+    // if you are using mapstruct in test code
+    annotationProcessor libraries["lombok-mapstruct-binding"]
```
Comment (Sgitario) [inline swatch-billable-usage/build.gradle:38]: I don't think we need the lombok-mapstruct-binding and `testAnnotationProcessor libraries["mapstruct-processor"]`.
I tried to remove them and the build worked fine (including tests).

**Review by Sgitario — CHANGES_REQUESTED** [review]
Apart from my comment about the dependencies, I'm also missing to cover these changes using junit tests in InternalBillableUsageResourceTest.

File: swatch-billable-usage/build.gradle (hunk 1 above)
Comment (kartikshahc) [reply swatch-billable-usage/build.gradle:38]: Interesting! The build is passing for me with it. But, I see we are not currently using it so, may as well remove it.

**Comment by kartikshahc** [comment]
> Apart from my comment about the dependencies, I'm also missing to cover these changes using junit tests in InternalBillableUsageResourceTest.

Yea the API is only meant for IQE so, I thought the IQE test coverage is sufficient but, I am okay adding this to our code base for better coverage.

File: swatch-billable-usage/build.gradle (hunk 1 above)
Comment (kartikshahc) [reply swatch-billable-usage/build.gradle]: Done

**Review by Sgitario — APPROVED** [review]

**Comment by kartikshahc** [comment]
@Aurobinda55 The test failures in CI seems unrelated to my changes. Can you please take a look if its the flaky tests that we discussed yesterday?

**Comment by Aurobinda55** [comment]
@kartikshahc  The API response looks good to me.

**Comment by Aurobinda55** [comment]
> @Aurobinda55 The test failures in CI seems unrelated to my changes. Can you please take a look if its the flaky tests that we discussed yesterday?

I checked the test locally and it passed so,  failed test looks unrelated for this PR. There is one card already to investigate the test failure.

**Review by Aurobinda55 — APPROVED** [review]

**Pull request merged**